docs/proposals/cluster-autoscaling.md

```diff
@@ -0,0 +1,269 @@
+this document will focus only on the CPU usage aspect (other metrics work in the exactly same way).
+The user is expected to provide the target level of utilization (which is now common for all metrics)
+and GCA will adjust the cluster size to meet the target. From the bird’s eye the architecture looks as below.
+
+```
+                  +--MIG-----------------------------+
+                  |                                  |
+                  | +--------+ +--------+ +--------+ |
+                  | |        | |        | |        | |
+                  | | Node1  | | Node2  | | NodeX  | |
+                  | |        | |        | |        | +<--+
+                  | +--------+ +--------+ +--------+ |   |
+                  | |cAdvisor| |cAdvisor| |cAdvisor| |   |
+                  | +--------+ +--------+ +----+---+ |   |
+                  |     |          |           |     |   |
+                  +-----|----------|-----------|-----+   |
+                        |          |           |         |
+                        |      +---v----+      |         |
+                        +----> |Heapster| <----+         |
+                               +---+----+                |
+                                   |                     |
+   Kubernetes                      |                     |
+-----------------------------------|---------------------|------
+   Google                          |                     |
+                                   |                     |
+                            +------v-----+       +-------+------+
+                            |Google Cloud|       |MIG Definition|
+                            | Monitoring |       +-------^------+
+                            +------------+               |
+                                   |                     |
+                                   |                     |
+                            +------v-----+               |
+                +------+    |Google Cloud+---------------+
+                |Config|--->| Autoscaler |
+                +------+    +------------+
+```
+
+CAdvisor instances gather metrics for nodes they are running on. Periodically Heapster (a service running
+on one of the nodes) queries all of the available nodes/cAdvisors, aggregates the metrics and pushes the
+usage to Google Cloud Monitoring (Heapster also takes Pod requests from apiserver but this fact is not
+depicted on the diagram). Google Cloud Autoscaler checks the metrics in GCM, compares the values
+and min/max cluster sizes from the config (that is passed through google-specific gcloud tool or web ui,
+and currently set up in kube-up/push) and updates the number of nodes in MIG (Managed Instance Group)
+definition. MIG then starts (or stops) the nodes.
+
+
+The current solution has a bunch of issues:
+
+* (I1) [GCE-specific API] Autoscaler configuration is Google-specific thus on other cloud providers the configuration may look completely different.
```

Comment:
change this to "[GCE-specific autoscaler configuration]" -- I think this is clearer than API